### PR TITLE
Fix CPU only for GPU tensors

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: src/pb_main_utils.cc [hunk 1]
```diff
@@ -25,11 +25,11 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <future>
+#include "pb_utils.h"
 #include "triton/backend/backend_common.h"
 #include "triton/core/tritonserver.h"
 
 #include "pb_main_utils.h"
```
Comment (CoderHam) [inline src/pb_main_utils.cc:32]: This should be on top, followed by empty line and then other includes.

File: src/pb_stub.cc [hunk 2]
```diff
@@ -47,10 +47,9 @@
...
 #include "shm_manager.h"
 
-#ifdef TRITON_ENABLE_GPU_TENSORS
-#include <cuda.h>
+#ifdef TRITON_ENABLE_GPU
 #include <cuda_runtime_api.h>
-#endif  // TRITON_ENABLE_GPU_TENSORS
+#endif  // TRITON_ENABLE_GPU
 
 #include "pb_stub.h"
```
Comment (CoderHam) [inline src/pb_stub.cc:54]: Same here

File: src/pb_tensor.cc [hunk 3]
```diff
@@ -459,24 +459,18 @@ PbTensor::AsNumpy() const
...
-    if (cuda_err != CUDA_SUCCESS) {
-      const char* error_string;
-      cuGetErrorString(cuda_err, &error_string);
-      throw PythonBackendException(
-          std::string(
-              "failed to get cuda pointer device attribute: " +
-              std::string(error_string))
-              .c_str());
-    }
+
```
Comment (CoderHam) [inline src/pb_tensor.cc:473]: Extra empty line

File: src/pb_tensor.h [hunk 4]
```diff
@@ -142,11 +142,16 @@ class PbTensor {
...
   static std::shared_ptr<PbTensor> LoadFromSharedMemory(
       std::unique_ptr<SharedMemory>& shm_pool, off_t tensor_offset);
-#ifdef TRITON_ENABLE_GPU_TENSORS
+#ifdef TRITON_ENABLE_GPU
   void SetReusedIpcHandle(cudaIpcMemHandle_t* cuda_ipc_mem_handle);
   void* GetGPUStartAddress();
   cudaIpcMemHandle_t* CudaIpcMemHandle();
-#endif  // TRITON_ENABLE_GPU_TENSORS
+
+  /// Get the GPU pointer offset.
```
Comment (CoderHam) [inline src/pb_tensor.h:150]: To be homogeneous add a similar comment above for  GetGPUStartAddress

File: src/pb_utils.cc [hunk 5]
```diff
@@ -341,4 +319,68 @@ FileExists(std::string& path)
   return stat(path.c_str(), &buffer) == 0;
 }
 
+#ifdef TRITON_ENABLE_GPU
+
+CUDADriverAPI::CUDADriverAPI()
+{
+  dl_open_handle_ = dlopen("libcuda.so", RTLD_LAZY);
+
+  // If dl_open_handle_ is succesfully opened, it must be able to find
```
Comment (CoderHam) [inline src/pb_utils.cc:328]: if libcuda is successfully opened. The handle cannot be opened.

**Review by CoderHam — CHANGES_REQUESTED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**